### DISCUSSION TITLE
add globalDependencies section

### DISCUSSION THREAD

**Discussion by its-dibo** [npm/feedback#336] [CLI]
add globalDependencies section to 'package.json', so we can automatically install all the required global packages by a particular project.

**use case:**
if you have an angular project and you manage to deploy it to firebase, you need to install '@angular/cli' and 'firebase' globally, in order to be able to run `ng` and `firebase` commands.

also, you may need to install some other packages globally such as 'typescript' to run `tsc` command.

if you moved your project to be deployed in another server, you may forget to install all the required global packages.

it is better to add `globalDependencies` section to the file 'package.json' to install all the required packages, including those global packages.

**Comment by ljharb** [comment]
This isn't the place for feature suggestions; that'd be https://github.com/npm/rfcs

However, there shouldn't ideally _be_ any such requirements. Those executable should be peer dependencies, so you install them in your top-level project, since ideally nothing should be globally installed at all.
